### PR TITLE
Scroll graduation branch rebased onto stabilization

### PR DESCRIPTION
**Pull request by vaoliva**
## Overview

This PR rebases the work from [graduation/scroll_view](https://github.com/microsoft/MixedRealityToolkit-Unity/tree/graduation/scroll_view) onto 2.5 stabilization using a clean branch not based on [event_propagation](https://github.com/microsoft/MixedRealityToolkit-Unity/tree/feature/event_propagation) branch.

All commits were previously approved with one exception:

 1. Test class was adjusted to adapt to latest stabilization changes found on BasePlaymodeTests.Setup()

_**Added comments on the bottom of the files with reference to respective PR approval**_ 

Example scenes and prefabs changes approved on #8380  but they will be substituted by new example scene from #8453.

### CODE REVIEW

File: Assets/MRTK/SDK/Features/UX/Interactable/Scripts/Interactable.cs [hunk 1]
```diff
@@ -679,11 +680,11 @@ protected virtual void OnDestroy()
...
         private void InternalUpdate()
         {
-            if (rollOffTimer < rollOffTime && HasPress)
+            if (rollOffTimer < RollOffTime && HasPress)
             {
                 rollOffTimer += Time.deltaTime;
 
-                if (rollOffTimer >= rollOffTime)
+                if (rollOffTimer >= RollOffTime)
                 {
```
Comment (vaoliva) [inline Assets/MRTK/SDK/Features/UX/Interactable/Scripts/Interactable.cs:688]: Change approved on #8091

File: Assets/MRTK/SDK/Experimental/ScrollingObjectCollection/ScrollingObjectCollection.cs [hunk 2]
```diff
@@ -2221,35 +2133,26 @@ void IMixedRealityTouchHandler.OnTouchUpdated(HandTrackingInputEventData eventDa
...
-                shouldSwallowEvents = false;
+                CurrentVelocityState = VelocityState.Calculating;
             }
         }
 
-        void IMixedRealityInputHandler.OnInputDown(InputEventData eventData) { }
-
         #endregion IMixedRealitySourceStateHandler implementation
     }
 }
```
Comment (vaoliva) [inline Assets/MRTK/SDK/Experimental/ScrollingObjectCollection/ScrollingObjectCollection.cs:2158]: Latest changes approved on #8430

File: Assets/MRTK/SDK/Experimental/Editor/Inspectors/ScrollingObjectCollection/ScrollingObjectCollectionInspector.cs [hunk 3]
```diff
@@ -43,322 +36,515 @@ public class ScrollingObjectCollectionInspector : UnityEditor.Editor
...
+            return StandardShaderUtility.IsUsingMrtkStandardShader(renderer.sharedMaterial) || renderer.sharedMaterial.shader == MRTKtmp;
         }
 
+        private void GetMouseViewportPosition()
+        {
+            mouseScreenPosition = HandleUtility.GUIPointToScreenPixelCoordinate(Event.current.mousePosition);
+            mouseScreenPosition = SceneView.currentDrawingSceneView.camera.ScreenToViewportPoint(mouseScreenPosition);
+        }
     }
 }
```
Comment (vaoliva) [inline Assets/MRTK/SDK/Experimental/Editor/Inspectors/ScrollingObjectCollection/ScrollingObjectCollectionInspector.cs:550]: Latest changes approved on #8430

File: Assets/MRTK/Examples/Experimental/ScrollingObjectCollection/Scripts/ScrollablePagination.cs [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        /// <summary>
+        /// Smoothly moves the scroll container a relative number of tiers of cells.
+        /// </summary>
+        public void ScrollByTier(int amount)
+        {
+            Debug.Assert(ScrollView != null, "Scroll view needs to be defined before using pagination.");
+            scrollView.MoveByTiers(amount);
+        }
+    }
+}
```
Comment (vaoliva) [inline Assets/MRTK/Examples/Experimental/ScrollingObjectCollection/Scripts/ScrollablePagination.cs:47]: Latest changes approved on #8430

File: Assets/MRTK/Examples/Experimental/ScrollingObjectCollection/Scripts/ScrollableListPopulator.cs [hunk 5]
```diff
@@ -144,38 +181,27 @@ public void MakeScrollingList()
...
-
-            g.transform.localPosition = Vector3.zero;
-            g.transform.localRotation = Quaternion.identity;
+            GameObject itemInstance = Instantiate(item, gridObjectCollection.transform);
+            itemInstance.SetActive(true);
         }
-
     }
-
 }
```
Comment (vaoliva) [inline Assets/MRTK/Examples/Experimental/ScrollingObjectCollection/Scripts/ScrollableListPopulator.cs:207]: Latest changes approved on #8430

File: Assets/MRTK/Tests/PlayModeTests/ScrollViewTests.cs [hunk 6]
```diff
@@ -0,0 +1,1419 @@
...
+
+            scrollObject.transform.position = position;
+            scrollObject.transform.rotation = rotation;
+
+            return scrollView;
+        }
+
+        #endregion Utilities
+    }
+}
```
Comment (vaoliva) [inline Assets/MRTK/Tests/PlayModeTests/ScrollViewTests.cs:1419]: Latest changes approved on #8430 + API change call to BasePlaymodeTests.Setup()

File: Assets/MRTK/Core/Inspectors/ClippingPrimitiveInspector.cs [hunk 7]
```diff
@@ -27,15 +27,31 @@ public abstract class ClippingPrimitiveEditor : UnityEditor.Editor
...
+                if (check.changed)
+                {
+                    // Flagging changes other than renderers
+                    clippingPrimitive.IsDirty = true;
+                }
+            }
+
             var currentRenderers = clippingPrimitive.GetRenderersCopy();
 
             // Add or remove and renderers that were added or removed via the inspector.
```
Comment (vaoliva) [inline Assets/MRTK/Core/Inspectors/ClippingPrimitiveInspector.cs:57]: Latest changes approved on #8424

File: Assets/MRTK/Core/Utilities/Rendering/MaterialInstance.cs [hunk 8]
```diff
@@ -177,6 +183,11 @@ private void Update()
...
 
         private void OnDestroy()
+        {
+            RestoreRenderer();
+        }
+
+        private void RestoreRenderer()
         {
             if (CachedRenderer != null && defaultMaterials != null)
             {
```
Comment (vaoliva) [inline Assets/MRTK/Core/Utilities/Rendering/MaterialInstance.cs:193]: Latest changes approved on #8424

File: Assets/MRTK/Core/Utilities/StandardShader/ClippingPrimitive.cs [hunk 9]
```diff
@@ -293,5 +312,14 @@ protected void ToggleClippingFeature(Material material, bool keywordOn)
...
+        private void CheckTransformChange()
+        {
+            if (transform.hasChanged)
+            {
+                IsDirty = true;
+                transform.hasChanged = false;
+            }          
+        }
     }
 }
```
Comment (vaoliva) [inline Assets/MRTK/Core/Utilities/StandardShader/ClippingPrimitive.cs:325]: Latest changes approved on #8424

**Review by MaxWang-MS — APPROVED** [review]
Looks good to me!

**Review by wiwei — APPROVED** [review]

**Comment by wiwei** [comment]
lgtm modulo the docs issue!

**Review by Cameron-Micka — APPROVED** [review]
Checked the clipping updates, looks good.

**Pull request merged**